### PR TITLE
Changes branch

### PR DESCRIPTION
**Pull request by hebaali9**
ضل بس اغير البروبس  رح اعملها بكرة حاولت فيها ورح ارجع اشوف
    // instead of sending props to the action, send the callback needed only. Use the same approach as in the register action

### CODE REVIEW

**Comment by hebaali9** [comment]
Hello still waiting your  comment

File: src/Actions/index.ts [hunk 1]
```diff
@@ -10,40 +10,13 @@ export const Update_Token_Action = function (token: string) {
   };
 };
 
-export const Update_Urls_List = function (urlsList: object[]) {
+export const UPDATE_URLS_LIST = function (urlsList: object[]) {
```
Comment (SirajKakeh) [inline src/Actions/index.ts:13]: array of Urls

File: README.md [hunk 2]
```diff
@@ -1 +1,10 @@
 # shortenUrl-typeScript
+
+- Folder names should start with a small letter
+- Components should be in a `components` folder
+- Remove unneeded comments
+- use `useSelector` to access global state properties (redux selectors)
+- typos & renames:
+    - `isLogedIn` => `isLoggedIn`
+    - `Sign_UP_Api` to `signup`
+    - `Inputurl` to `UrlInput`
```
Comment (SirajKakeh) [inline README.md:10]: remove comments

File: src/ProtectedRoutes.tsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
 
 function ProtectedRoutes({ children }: { children: JSX.Element }) {
-  const token = store.getState().token;
+  interface RootState {
+    token: string;
+  }
+
+  const selectToken = (state: RootState) => state.token;
+
+  const token = useSelector(selectToken);
```
Comment (SirajKakeh) [inline src/ProtectedRoutes.tsx:11]: let's use the arrow function directly here, no need for extra variable

File: src/app/App.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import ProtectedRoutes from "../ProtectedRoutes";
+import SignUp from "../signUp/SignUp";
+import ShortenUrlPage from "../shortenUrl/ShortenUrlPage";
+import ListOfUrl from "../listOfUrls/ListOfUrl";
+import { store } from "../store";
+
+function App() {
+  const token = store.getState().token;
```
Comment (SirajKakeh) [inline src/app/App.tsx:9]: let's use selectors

File: src/inputUrl/Inputurl.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import "./InputURL.css";
+import { store } from "../store";
+import { Input_Url_Api } from "../actions";
+import { useSelector } from "react-redux";
+
+function UrlInput(props: any) {
```
Comment (SirajKakeh) [inline src/inputUrl/Inputurl.tsx:7]: let's change file name as well

File: src/listOfUrls/ListOfUrl.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import PopUp from "../popup/PopUp";
+import { store } from "../store";
+import { editUrl, On_Delete_Api, getUrlsList } from "../actions";
+import { Url } from "../types/types";
+import { useSelector } from "react-redux";
+
+function ListOfUrl() {
+  const [isPopUpOpen, setIsPopUpOpen] = useState(false);
+  const [selectedUrl, setSelectedUrl] = useState<Url | null>(null);
+  interface RootState {
```
Comment (SirajKakeh) [inline src/listOfUrls/ListOfUrl.tsx:17]: use types and make it outside the component

File: src/signUp/SignUp.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            type="radio"
+            value="F"
+            name="gender"
+            onChange={(e) => {
+              setGender(e.target.value as Gender);
+            }}
+          />
+          Female
+        </p>
+        {/* ?? */}
```
Comment (SirajKakeh) [inline src/signUp/SignUp.tsx:108]: let's delete this

**Review by SirajKakeh — APPROVED** [review]

File: src/Actions/index.ts (hunk 1 above)
Comment (hebaali9) [reply src/Actions/index.ts:13]: Did not got it 
tried but did not worked

**Comment by hebaali9** [comment]
in UrlInput file 
I changed the type of the props but got an error 

Screenshot 2022-12-10 at 2 23 52 AM

**Pull request merged**